daal4py/sklearn/neighbors/knn.py

```diff
@@ -163,6 +202,65 @@ def kneighbors(self, X=None, n_neighbors=None, return_distance=True):
 
 
 class KNeighborsClassifier(BaseKNeighborsClassifier, KNeighborsMixin):
+    def fit(self, X, y):
+        X_incorrect_type = isinstance(X, NeighborsBase) or isinstance(X, BallTree) or isinstance(X, KDTree)
+
+        if not X_incorrect_type:
+            X, y = self._validate_data(X, y, accept_sparse="csr", multi_output=True)
```

Comment:
Verify that `_validate_data` is not hotspot when pandas DataFrame/Series is passed. It might be better to handle these formats in another way.

Reply:
I checked it. `_validete_data` is about 1-3% from fitting time. So, I think we don't need to change something